app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -222,15 +235,60 @@ public void onGetUserAccount(UserAccount userAccount) {
 
     }
 
-    private void openNewSurvey(Activity activity) {
+    private void openNewSurvey(final Activity activity) {
+        Survey survey = createNewConnectSurvey();
+
+        IMainExecutor mainExecutor = new UIThreadExecutor();
+        IAsyncExecutor asyncExecutor = new AsyncExecutor();
+        ISurveyRepository surveyRepository = new SurveyLocalDataSource();
+        SaveSurveyUseCase saveSurveyUseCase = new SaveSurveyUseCase(asyncExecutor, mainExecutor, surveyRepository);
+
+        saveSurveyUseCase.execute(survey, new SaveSurveyUseCase.Callback() {
+            @Override
+            public void onSurveySaved(Survey survey) {
+                saveValuesFromIntent(activity, survey);
+                if(!DashboardActivity.dashboardActivity.isPreLoadSurveyOpenning()) {
+                    mDashboardActivity.initSurvey();
+                }
+            }
+        });
+    }
+
+    private Survey createNewConnectSurvey() {
         ProgramDB programDB = ProgramDB.findById(PreferencesEReferral.getUserProgramId());
-        // Put new survey in session
-        SurveyDB survey = new SurveyDB(null, programDB, Session.getUserDB());
-        survey.save();
-        Session.setMalariaSurveyDB(survey);
-        //Look for coordinates
-        prepareLocationListener(activity, survey);
-        mDashboardActivity.initSurvey();
+        UserAccount userAccount = new UserAccount(Session.getUserDB().getName(),
```

Comment:
can you refactor this for GetUserUserAccountUseCase instead of use session for create UserAccount?